### PR TITLE
Resolve Issue 383

### PR DESCRIPTION
**Pull request by stevedum**
change new database message to
No Lifelines database found.
Do you wnat to create a database in <path/name>

also other changes
   update xterm_decode
some clean up on test suite
Results: all tests pass except for string/string-unicode
    Except requires previous version of yacc.y as current version
    crashes in multiple places with double free issues.

### CODE REVIEW

File: src/gedlib/messages.c [hunk 1]
```diff
@@ -44,8 +44,8 @@ STRING empstr71 = SS "
 ZST qSiddbse      = N_("You must identify a database.");
 ZST qSidldir      = N_("What directory holds (or will hold) the LifeLines database? (? to list)");
 ZST qSidldrp      = N_("enter path: ");
-ZST qSnodbse      = N_("There is no LifeLines database in that directory.");
-ZST qScrdbse      = N_("Do you want to create a database there?");
+ZST qSnodbse      = N_("No LifeLines database found.");
+ZST qScrdbse      = N_("Do you want to create a database at");
```
Comment (memmerto) [inline src/gedlib/messages.c]: You can use printf specifiers in these translated strings.

```
ZST qScrdbse = N_("Do you want to create a database at '%s'?");
```

File: src/liflines/selectdb.c [hunk 2]
```diff
@@ -181,7 +182,8 @@ open_or_create_database (INT alteration, STRING *dbused)
 	strupdate(dbused, dbdir);
 
 	/* Is user willing to make a new db ? */
-	if (!ask_yes_or_no_msg(_(qSnodbse), _(qScrdbse))) 
+        snprintf(newmsg,sizeof(newmsg),"%s '%s' ?",qScrdbse,*dbused);
```
Comment (memmerto) [inline src/liflines/selectdb.c]: Then this becomes:

```
snprintf(newmsg,sizeof(newmsg),qScrdbse,*dbused);
```

Doing it this way ensures that when the message is translated, the %s can get put into the right spot.  Also knowing that the message is a question is important, so the question mark should be in the message, not the snprintf here.

File: src/gedlib/messages.c (hunk 1 above)
Comment (stevedum) [reply src/gedlib/messages.c]: thanks, done

**Comment by memmerto** [comment]
Included in #385

**Pull request closed without merging**